examples/matching-engine/tests/transaction.rs

```diff
@@ -0,0 +1,263 @@
+            fungible_bytecode_id_b,
+            params_b,
+            initial_state_b.build(),
+            vec![],
+        )
+        .await;
+
+    fungible_chain_a.register_application(token_id_b).await;
+    fungible_chain_b.register_application(token_id_a).await;
+
+    for (owner, amount) in [
+        (admin_account, None),
+        (owner_a, Some(Amount::from_tokens(10))),
+        (owner_b, None),
+    ] {
+        let value = FungibleTokenAbi::query_account(token_id_a, &fungible_chain_a, owner).await;
+        assert_eq!(value, amount);
+    }
+    for (owner, amount) in [
+        (admin_account, None),
+        (owner_a, None),
+        (owner_b, Some(Amount::from_tokens(9))),
+    ] {
+        let value = FungibleTokenAbi::query_account(token_id_b, &fungible_chain_b, owner).await;
+        assert_eq!(value, amount);
+    }
+
+    let tokens = [token_id_a, token_id_b];
+    let matching_parameter = Parameters { tokens };
+    let matching_id = matching_chain
+        .create_application::<MatchingEngineAbi>(
+            bytecode_id,
+            matching_parameter,
+            (),
+            vec![token_id_a.forget_abi(), token_id_b.forget_abi()],
+        )
+        .await;
+    // Doing the registrations
+    matching_chain.register_application(token_id_a).await;
+    matching_chain.register_application(token_id_b).await;
+    fungible_chain_a.register_application(matching_id).await;
+    fungible_chain_b.register_application(matching_id).await;
+
+    // Creating the bid orders
+    let mut orders_bids = Vec::new();
+    for price in [1, 2] {
+        let price = Price { price };
+        let order = Order::Insert {
+            owner: owner_a,
+            amount: Amount::from_tokens(3),
+            nature: OrderNature::Bid,
+            price,
+        };
+        let operation = Operation::ExecuteOrder { order };
+        let order_messages = fungible_chain_a
+            .add_block(|block| {
+                block.with_operation(matching_id, operation);
+            })
+            .await;
+
+        assert_eq!(order_messages.len(), 3);
+        orders_bids.extend(order_messages);
+    }
+
+    matching_chain
+        .add_block(|block| {
+            block.with_incoming_messages(orders_bids);
+        })
+        .await;
+
+    fungible_chain_a.handle_received_messages().await;
+    fungible_chain_b.handle_received_messages().await;
+    matching_chain.handle_received_messages().await;
+
+    // Checking the values for token_a
+    for (owner, amount) in [
+        (admin_account, None),
+        (owner_a, Some(Amount::from_tokens(1))),
+        (owner_b, None),
+    ] {
+        let value = FungibleTokenAbi::query_account(token_id_a, &fungible_chain_a, owner).await;
+        assert_eq!(value, amount);
+    }
+    for (owner, amount) in [(admin_account, None), (owner_a, None), (owner_b, None)] {
+        assert_eq!(
+            FungibleTokenAbi::query_account(token_id_a, &fungible_chain_b, owner).await,
```

Comment:
Is `token_id_a` even registered on `fungible_chain_b` yet?

Reply:
No, but the test passes. And notice that all the results are `None`.